source/JsMaterialX/test/XmlIo.spec.js

```diff
@@ -1,20 +1,243 @@
+    // Should we use a pre-test script that fetches the files and makes them available somehow?
+    const libraryFilenames = ['stdlib_defs.mtlx', 'stdlib_ng.mtlx'];
+    const examplesWithIncludes = [
+        'Looks.mtlx',
+        'PaintMaterials.mtlx',
+        'PostShaderComposite.mtlx',
+    ];
+    const examplesWithoutIncludes = [
+        'CustomNode.mtlx',
+        'GeomInfos.mtlx',
+        'MaterialBasic.mtlx',
+        'MultiOutput.mtlx',
+        'NodeGraphs.mtlx',
+        'PreFlattenedGraph.mtlx',
+        'PreShaderComposite.mtlx',
+        'SimpleSrf.mtlx',
+        'SubGraphs.mtlx',
+    ];
+    const exampleFilenames = examplesWithIncludes.concat(examplesWithoutIncludes);
+
+    async function readStdLibrary(asString = false) {
+        const libs = [];
+        let iterable = libraryFilenames;
+        if (asString) {
+            const libraryMtlxStrings = getMtlxStrings(libraryFilenames, libraryPath);
+            iterable = libraryMtlxStrings;
+        }
+        for (let file of iterable) {
+            const lib = mx.createDocument();
+            if (asString) {
+                await mx.readFromXmlString(lib, file, libraryPath);
+            } else {
+                await mx.readFromXmlFile(lib, file, libraryPath);
+            }
+            libs.push(lib);
+        };
+        return libs;
+    }
+
+    async function readAndValidateExamples(examples, libraries, readFunc, searchPath = undefined) {
+        for (let file of examples) {
+            const doc = mx.createDocument();
+            await readFunc(doc, file, searchPath);
+            // Import stdlib into the current document and validate it.
+            for (let lib of libraries) {
+                doc.importLibrary(lib);
+            }
+            expect(doc.validate()).to.be.true;
+
+            // Make sure the document does actually contain something.
+            let valueElementCount = 0;
+            const treeIter = doc.traverseTree();
+            for(const elem of treeIter) {
+                if (elem instanceof mx.ValueElement) {
+                    valueElementCount++;
+                }
+            }
+            expect(valueElementCount).to.be.greaterThan(0);
+        };
+    }
+
+    before(async () => {
+        mx = await Module();
+    });
+
+    it('Read XML without includes from file', async () => {
+        // Read the standard library
+        const libs = await readStdLibrary(false);
+
+        // Read and validate the example documents.
+        readAndValidateExamples(examplesWithoutIncludes, libs,
+            async (document, file, sp) => {
+                await mx.readFromXmlFile(document, file, sp);
+            }, examplesPath);
+
+        // Read the same document twice, and verify that duplicate elements
+        // are skipped.
+        const doc = mx.createDocument();
+        const filename = 'CustomNode.mtlx';
+        await mx.readFromXmlFile(doc, filename, examplesPath);
```

Comment:
not sure what I'm expecting from this? So, if I read the same file several times, do you get the document as it was before? Or what is added? if nothing is added can we add an assert more explicit, like the exported string is the same of the number of nodedefs are preserved?